sensor/kubernetes/listener/resources/secrets_test.go

```diff
@@ -369,3 +371,62 @@ func getImageIntegrationEvents(events *component.ResourceEvent) []*central.Senso
 
 	return iiEvents
 }
+
+// TestSkipIntegrationCreate verifies that integrations are NOT created in
+// various condition.
+func TestSkipIntegrationCreate(t *testing.T) {
+	globalPullSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      openshift.GlobalPullSecretName,
+			Namespace: openshift.GlobalPullSecretNamespace,
+		},
+	}
+	globalPullSecretDupe := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "installation-pull-secrets",
+			Namespace: "openshift-image-registry",
+		},
+	}
+	someOtherSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "stackrox",
+			Namespace: "stackrox",
+		},
+	}
+
+	t.Run("do not skip when all relevant features enabled", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SourcedAutogeneratedIntegrations, true)
+		t.Setenv(env.AutogenerateGlobalPullSecRegistries.EnvVar(), strconv.FormatBool(true))
```

Comment:
think I saw this before, but another super nit: you can just do `"true"`. Guess it's just personal preference

Reply:
I changed this based on a past comment:  https://github.com/stackrox/stackrox/pull/8499#discussion_r1393652964